contracts/AelinUpFrontDeal.sol

```diff
@@ -257,43 +262,48 @@ contract AelinUpFrontDeal is AelinERC20, MinimalProxyFactory, IAelinUpFrontDeal
             // Claim Deal Tokens
             bool deallocate = totalPoolShares > _underlyingDealTokenTotal;
+            uint256 adjustedShareAmountForUser;
+            uint256 precisionAdjustedRefund;
 
             if (deallocate) {
                 // adjust for deallocation and mint deal tokens
-                uint256 adjustedDealTokensForUser = (((poolSharesPerUser[msg.sender] * _underlyingDealTokenTotal) /
-                    totalPoolShares) * (BASE - AELIN_FEE - dealData.sponsorFee)) / BASE;
+                adjustedShareAmountForUser =
+                    (((poolSharesPerUser[msg.sender] * _underlyingDealTokenTotal) / totalPoolShares) *
+                        (BASE - AELIN_FEE - dealData.sponsorFee)) /
+                    BASE;
                 poolSharesPerUser[msg.sender] = 0;
 
                 // refund any purchase tokens that got deallocated
                 uint256 purchasingRefund = purchaseTokensPerUser[msg.sender] -
                     ((purchaseTokensPerUser[msg.sender] * _underlyingDealTokenTotal) / totalPoolShares);
-                purchaseTokensPerUser[msg.sender] = 0;
 
-                uint256 precisionAdjustedRefund = purchasingRefund > IERC20(_purchaseToken).balanceOf(address(this))
+                precisionAdjustedRefund = purchasingRefund > IERC20(_purchaseToken).balanceOf(address(this))
                     ? IERC20(_purchaseToken).balanceOf(address(this))
                     : purchasingRefund;
 
-                // mint deal tokens and transfer purchase token refund
-                _mint(msg.sender, adjustedDealTokensForUser);
+                // Transfer purchase token refund
                 IERC20(_purchaseToken).safeTransfer(msg.sender, precisionAdjustedRefund);
-
-                emit ClaimDealTokens(msg.sender, adjustedDealTokensForUser, precisionAdjustedRefund);
             } else {
                 // mint deal tokens when there is no deallocation
-                uint256 adjustedDealTokensForUser = ((BASE - AELIN_FEE - dealData.sponsorFee) *
-                    poolSharesPerUser[msg.sender]) / BASE;
-                poolSharesPerUser[msg.sender] = 0;
-                purchaseTokensPerUser[msg.sender] = 0;
-                _mint(msg.sender, adjustedDealTokensForUser);
-                emit ClaimDealTokens(msg.sender, adjustedDealTokensForUser, 0);
+                adjustedShareAmountForUser =
+                    ((BASE - AELIN_FEE - dealData.sponsorFee) * poolSharesPerUser[msg.sender]) /
+                    BASE;
             }
+            poolSharesPerUser[msg.sender] = 0;
+            purchaseTokensPerUser[msg.sender] = 0;
+
+            // mint vesting token
+            _mint(msg.sender, tokenCount);
+            tokenDetails[tokenCount] = TokenDetails(adjustedShareAmountForUser, block.timestamp);
+            emit ClaimVestingToken(msg.sender, tokenCount, adjustedShareAmountForUser, precisionAdjustedRefund);
```

Comment:
again, I agree we could name it better but I think changing the name here is bad and makes the subgraph and code in general more complicated. We can debate this if you want

Reply:
I guess with this change we are adding new arguments. one idea is that we could keep the old event but add a second event on top with the new data. We can make the subgraph changes here though since it is a major change. I just prefer to keep the subgraph simple as much as possible

Reply:
yes so basically the logic in this section looks good but I think we can do better with events in general to make the subgraph less messy